### PR TITLE
Updated OES Images and creating New PR.

### PR DESCRIPTION
**Pull request by vkvamsiopsmx**

### CODE REVIEW

File: charts/oes/templates/oes-redis-statefulset.yaml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        - name: REDIS_REPLICATION_MODE
+          value: master
+        - name: REDIS_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              key: redis-password
+              name: redis-secret
+        - name: REDIS_DISABLE_COMMANDS
+          value: FLUSHDB,FLUSHALL
+        image: docker.io/bitnami/redis:4.0.11-debian-9
```
Comment (Nirmalyasen) [inline charts/oes/templates/oes-redis-statefulset.yaml:31]: Let's externalize the image

File: charts/oes/templates/oes-sapor-cacerts.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
```
Comment (Nirmalyasen) [inline charts/oes/templates/oes-sapor-cacerts.yaml:1]: what is the purpose of this one?

File: charts/oes/templates/oes-sapor-deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            port: 8085
+          initialDelaySeconds: 180
+          periodSeconds: 60
+      imagePullSecrets:
+      - name: regcred
+      volumes:
+        - name: api-volume
+          configMap:
+            name: sapor-config
+        - name: certs-volume
```
Comment (Nirmalyasen) [inline charts/oes/templates/oes-sapor-deployment.yaml:50]: Why is this mandatory and not optionally overwritten by the end user if they have custom cacerts? Why are we not bundling the default cacerts that java uses?

File: charts/oes/templates/oesdb-deployment.yaml [hunk 4]
```diff
@@ -1,11 +1,16 @@
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: apps/v1
+kind: StatefulSet
 metadata:
   labels:
     app: db-opsmx
   name: db-opsmx
 spec:
+  podManagementPolicy: {{ .Values.opsmxdb.podManagementPolicy }}
```
Comment (Nirmalyasen) [inline charts/oes/templates/oesdb-deployment.yaml:8]: Specially for things like this, we should have defaults that we think will work 90% of the cases and the user should provide override if they want something different.

File: charts/oes/templates/oesdb-deployment.yaml [hunk 5]
```diff
@@ -14,20 +19,32 @@ spec:
...
       containers:
-      - image: {{ .Values.OESDBImage }}
-        imagePullPolicy: IfNotPresent
+      - image: {{ .Values.opsmxdb.image.repository }}:{{ .Values.opsmxdb.image.tag }}
+        imagePullPolicy: {{ .Values.opsmxdb.image.pullPolicy }}
         name: db-opsmx
         ports:
         - containerPort: 5432
           protocol: TCP
-      {{- if .Values.persistence.enabled }}
```
Comment (Nirmalyasen) [inline charts/oes/templates/oesdb-deployment.yaml:23]: This should be an optional flag that a user can turn off if at all needed. This is seen in minio and redis charts and we should do the same.

File: charts/oes/templates/opa-deploy.yaml [hunk 6]
```diff
@@ -0,0 +1,26 @@
+apiVersion: extensions/v1beta1
```
Comment (Nirmalyasen) [inline charts/oes/templates/opa-deploy.yaml:1]: we should made this conditional...on by default and the user can turn if off if needed.

File: charts/oes/templates/opa-deploy.yaml [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  template:
+    metadata:
+      labels:
+        app: oes
+        component: opa
+      name: opa
+    spec:
+      containers:
+        - name: opa
+          image: openpolicyagent/opa:latest
```
Comment (Nirmalyasen) [inline charts/oes/templates/opa-deploy.yaml:23]: externalize

File: charts/oes/templates/ops-svc.yaml [hunk 8]
```diff
@@ -0,0 +1,13 @@
+apiVersion: v1
```
Comment (Nirmalyasen) [inline charts/oes/templates/ops-svc.yaml:1]: should this file be called ops-svc or opa-svc? should be conditional

File: charts/oes/values.yaml [hunk 9]
```diff
@@ -1,20 +1,80 @@
...
+      repository: opsmx11/oes-ui
+      tag: v0.202007011835
+      pullPolicy: IfNotPresent
+  config:
+    db:
+      dbUrl: jdbc:postgresql://db-opsmx:5432/opsmx
+      username: postgres
+      password: networks123
+      #backendPort: 8090
+    server:
```
Comment (Nirmalyasen) [inline charts/oes/values.yaml:42]: why is this needed here?

File: charts/oes/values.yaml [hunk 10]
```diff
@@ -1,20 +1,80 @@
...
+      baseUrl: http://sapor-svc:8085
+      enabled: true
+    spinnaker:
+      oesGateURL: http://106.51.137.146:31174/  # Value of the OES-GATE URL
+      oesUIcors: ^https?://(?:localhost|oes-ui-svc|opsmx.com)(?::[1-9]\d*)?/? # Value of the OES UI URL Regex
```
Comment (Nirmalyasen) [inline charts/oes/values.yaml:47]: why is this needed?

File: charts/oes/values.yaml [hunk 11]
```diff
@@ -1,20 +1,80 @@
...
+    db:
+      dbUrl: jdbc:postgresql://db-opsmx:5432/opsmx
+      username: postgres
+      password: networks123
+      #backendPort: 8090
+    server:
+      baseUrl: http://sapor-svc:8085
+      enabled: true
+    spinnaker:
+      oesGateURL: http://106.51.137.146:31174/  # Value of the OES-GATE URL
```
Comment (Nirmalyasen) [inline charts/oes/values.yaml:46]: why is this needed here?

File: charts/oes/values.yaml [hunk 12]
```diff
@@ -1,20 +1,80 @@
...
+      oesUIcors: ^https?://(?:localhost|oes-ui-svc|opsmx.com)(?::[1-9]\d*)?/? # Value of the OES UI URL Regex
+      spinGateURL: http://106.51.137.146:31174/gate # Value of the Spinnaker-gate URL
+      spinExternalGateURL: http://106.51.137.146:31174/gate # Value of the Spinnaker URL to access spinnaker from UI
+      spinuser: dummyuser # Value of the Spinnaker Username (Deck)
+      spinpasswd: dummypwd # Value of the Spinnaker Password (Deck)
+      adminLoginEnabled: false
+    ldap:
+      enabled: true
```
Comment (Nirmalyasen) [inline charts/oes/values.yaml:54]: say we do not install ldap but configure with existing authentication - how will that be handled?

File: charts/oes/values.yaml [hunk 13]
```diff
@@ -1,20 +1,80 @@
...
+      basedn: dc=example,dc=org # Value of ldap.base.dn
+      pattern: (&(cn=USERNAME)) # Value of ldap.user.filter.pattern
+        # Password for the admin user
+      adminPassword: dummypwd
+      omitClusterIP: true
+###################################################
+opsmxdb:
+  # Change this to false if autopilot is installed along with OES
+  # as autopilot can use the DB of OES
+  #enabled: true
```
Comment (Nirmalyasen) [inline charts/oes/values.yaml:66]: did we mean to keep this commented?

File: charts/oes/templates/oes-redis-statefulset.yaml (hunk 1 above)
Comment (abhinayopsmx) [reply charts/oes/templates/oes-redis-statefulset.yaml:31]: Done

File: charts/oes/values.yaml (hunk 9 above)
Comment (abhinayopsmx) [reply charts/oes/values.yaml:42]: I guess to allow oes-gate reach oes-server(sapor), isn't it?

File: charts/oes/config/spinnaker.yaml [hunk 14]
```diff
@@ -0,0 +1,97 @@
+## WARNING
```
[the remark this answers is not in the document]
Comment (vkvamsiopsmx) [reply charts/oes/config/spinnaker.yaml:1]: This is kept to handle the redis for OES Gate, but  now its removed and the same is handled in gate.yaml.

File: charts/oes/values.yaml (hunk 13 above)
Comment (abhinayopsmx) [reply charts/oes/values.yaml:66]: No, missed to uncomment. Fixed it.

File: charts/oes/values.yaml (hunk 11 above)
Comment (abhinayopsmx) [reply charts/oes/values.yaml:46]: This URL is used by OES-UI to reach OES-GATE.

File: charts/oes/templates/oes-gate-cm.yaml [hunk 15]
```diff
@@ -0,0 +1,20 @@
+apiVersion: v1
```
[the remark this answers is not in the document]
Comment (vkvamsiopsmx) [reply charts/oes/templates/oes-gate-cm.yaml:1]: Yes, we have made it to a Single folder and updated the chart. Now config map will pick all the files from that folder.

I don't believe the code uses spinnakerconfig.yml.vault.::: Yes it is removed now, as it is not needed.

File: charts/oes/values.yaml (hunk 10 above)
Comment (vkvamsiopsmx) [reply charts/oes/values.yaml:65]: oesGateURL: OES Gate URL is being used OES-UI.

oesUIcors: This regex is used by OES-Gate, to trust only the OES-UI URL and reject the cross origin URL requests.

File: charts/oes/templates/ops-svc.yaml (hunk 8 above)
Comment (vkvamsiopsmx) [reply charts/oes/templates/ops-svc.yaml:1]: Done.

File: charts/oes/templates/oesdb-deployment.yaml (hunk 4 above)
Comment (abhinayopsmx) [reply charts/oes/templates/oesdb-deployment.yaml]: Shall we keep default value as "Parallel" instead of "OrderedReady" ?

File: charts/oes/templates/oesdb-deployment.yaml (hunk 5 above)
Comment (vkvamsiopsmx) [reply charts/oes/templates/oesdb-deployment.yaml:23]: Now as we have moved to stateful set instead of the deployment, this persistence case no longer exists.

As a whole we have added a condition, that only if the user wants to use opsmx database then he can set the flag true in values.yaml.

File: charts/oes/templates/opa-deploy.yaml (hunk 6 above)
Comment (vkvamsiopsmx) [reply charts/oes/templates/opa-deploy.yaml:1]: Done.

File: charts/oes/templates/opa-deploy.yaml (hunk 7 above)
Comment (vkvamsiopsmx) [reply charts/oes/templates/opa-deploy.yaml:23]: Done.

File: charts/oes/values.yaml (hunk 12 above)
Comment (vkvamsiopsmx) [reply charts/oes/values.yaml:73]: If LDAP Enabled is set to true, user need to provide their own ldap details in the values.yaml under ldap section.So that oue OES-Gate will take the user's LDAP Configuration for the authentication.

File: charts/oes/templates/oes-sapor-cacerts.yaml (hunk 2 above)
Comment (vkvamsiopsmx) [reply charts/oes/templates/oes-sapor-cacerts.yaml:1]: Our Sapor Server is expecting the ca cert of the spinnaker URL (If SSL is enabled for the Spinnaker). By default it is not getting added to cacert of the Sapor. So we have taken out the existing sapor cacerts file and appended the ca cert of our demo.opsmx.com and giving it to the pod as a config map.

File: charts/oes/templates/oes-gate-cm.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (abhinayopsmx) [reply charts/oes/templates/oes-gate-cm.yaml:1]: Found a way to fix it, please check my latest commit in your repo.

File: charts/oes/templates/opa-deploy.yaml [hunk 16]
```diff
@@ -0,0 +1,30 @@
+{{- if .Values.opa.enabled }}
+apiVersion: extensions/v1beta1
```
Comment (Nirmalyasen) [inline charts/oes/templates/opa-deploy.yaml:2]: Should we update the apiVersion to apps/v1?

File: charts/opsmx-autopilot/templates/autopilot-deploy.yaml [hunk 17]
```diff
@@ -1,35 +1,39 @@
 apiVersion: extensions/v1beta1
```
Comment (Nirmalyasen) [inline charts/opsmx-autopilot/templates/autopilot-deploy.yaml:1]: Should this be updated to apps/v1?

File: charts/oes/templates/opa-deploy.yaml (hunk 16 above)
Comment (abhinayopsmx) [reply charts/oes/templates/opa-deploy.yaml:2]: Yes, updated the same

File: charts/opsmx-autopilot/templates/autopilot-deploy.yaml (hunk 17 above)
Comment (abhinayopsmx) [reply charts/opsmx-autopilot/templates/autopilot-deploy.yaml:1]: Yes, updated

**Comment by vkvamsiopsmx** [comment]
Closing this PR, as this has the merge conflicts with the Sagay changes, a new PR has be generated which do not have any merge conflicts.
New PR:https://github.com/OpsMx/enterprise-spinnaker/pull/18 @Nirmalyasen  can you please review this updated PR.
